clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3556,6 +3573,20 @@ void Sema::AddSYCLIntelMaxWorkGroupSizeAttr(Decl *D,
   if (!XDim || !YDim || !ZDim)
     return;
 
+  // If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
+  // a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
+  // attribute, check to see if values of reqd_work_group_size arguments are
+  // equal or less than values of max_work_group_size attribute arguments.
+  if (const auto *DeclAttr = D->getAttr<ReqdWorkGroupSizeAttr>()) {
+    if (checkWorkGroupSizeAttrValues(DeclAttr->getXDim(), XDim),
+	checkWorkGroupSizeAttrValues(DeclAttr->getYDim(), YDim),
+	checkWorkGroupSizeAttrValues(DeclAttr->getZDim(), ZDim)) {
+      Diag(CI.getLoc(), diag::err_conflicting_sycl_function_attributes)
+         << CI << DeclAttr->getSpelling();
```

Comment:
```suggestion
         << CI << DeclAttr;
```

Reply:
We should also note the conflicting attribute, right? (Same question below).

Reply:
> We should also note the conflicting attribute, right? (Same question below).

It was not added with original implementation. So i did not do it.
I will add a note. Thanks for the suggestion.

Reply:
>>We should also note the conflicting attribute, right? (Same question below).
Done

Reply:
The diagnostic also needs to be updated to drop it's manual adding of the single quotes.

Reply:
Done